### PR TITLE
Limit logs and prevent full API response logging

### PR DESCRIPTION
**Pull request by corydhmiller**
This PR rewrites the full writing of the API response to the `logs.txt` file and instead just returns a success log.

This also prevents more than 1,000 lines from being loaded into the debug console.

### CODE REVIEW

File: admin/section/class-convertkit-settings-general.php [hunk 1]
```diff
@@ -166,7 +166,29 @@ public function register_fields() {
...
+			$this->name
+		);
 
+		add_settings_field(
+			'debug',
+			'Debug',
+			array( $this, 'debug_callback' ),
+			$this->settings_key,
+			$this->name
+		);
```
Comment (bradpauly) [inline admin/section/class-convertkit-settings-general.php:191]: It looks like this is called the same way three times. Are they all needed? https://github.com/ConvertKit/convertkit-wordpress/blob/211a0b3fe6b9cfc522e940d0fef9248cebb38150/admin/section/class-convertkit-settings-general.php#L177-L198

File: admin/section/class-convertkit-settings-tools.php [hunk 2]
```diff
@@ -67,9 +67,22 @@ public function render() {
...
+				$log .= 'End of file reached.';
+				break;
+			}
+			if ($i === $log_limit - 1) {
+				$log .= '--- 1,000 lines reached, end of log print. Click "Clear Log" if your logs are taking too long to load. ---';
+				break;
+			}
+			$log .= fgets($fp);
+		}
+		fclose($fp);
```
Comment (bradpauly) [inline admin/section/class-convertkit-settings-tools.php:85]: 👍

**Review by bradpauly — APPROVED** [review]

File: includes/class-convertkit-api.php [hunk 3]
```diff
@@ -114,7 +114,7 @@ public function update_resources( $api_key, $api_secret ) {
 
 		// Forms and Landing Pages.
 		$api_response = $this->_get_api_response( 'forms' );
-		WP_ConvertKit::log( 'API Response: ' . print_r( $api_response, true ) );
+		WP_ConvertKit::log( 'API Response: Success (Returned ' . count($api_response['forms']) . " forms.)");
```
Comment (marcboquet) [inline includes/class-convertkit-api.php:117]: Will this work as expected if `$api_response['forms']` is null, or if `$api_response` is not an object?

File: admin/section/class-convertkit-settings-general.php (hunk 1 above)
Comment (corydhmiller) [reply admin/section/class-convertkit-settings-general.php:184]: This was a mistake, this should be fixed now.

**Comment by corydhmiller** [comment]
@marcboquet @bradpauly One mini fix for file detection, other than that we're good to push for 1.9.5. Once I get 👍 I'll start the process.

**Review by marcboquet — APPROVED** [review]

**Pull request merged**